delta-lake/delta-spark350db143/src/main/scala/com/databricks/sql/transaction/tahoe/rapids/GpuDeltaDataSource.scala

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * This file was derived from DeltaDataSource.scala in the
+ * Delta Lake project at https://github.com/delta-io/delta.
+ *
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.delta
+
+import com.databricks.sql.transaction.tahoe.{DeltaConfigs, DeltaErrors, DeltaOptions}
+import com.databricks.sql.transaction.tahoe.commands.WriteIntoDeltaEdge
+import com.databricks.sql.transaction.tahoe.rapids.{GpuDeltaLog, GpuWriteIntoDelta}
+import com.databricks.sql.transaction.tahoe.sources.{DeltaDataSource, DeltaSourceUtils}
+import com.nvidia.spark.rapids.{GpuCreatableRelationProvider, RapidsConf}
+
+import org.apache.spark.sql.{DataFrame, SaveMode, SQLContext}
+import org.apache.spark.sql.sources.BaseRelation
+
+/** GPU version of DeltaDataSource from Delta Lake. */
+class GpuDeltaDataSource(rapidsConf: RapidsConf) extends GpuCreatableRelationProvider {
```

Comment:
Am I right in understanding that `GpuDeltaDataSource` and `GpuWriteIntoDelta` are identical across Databricks versions, including 14.3?  I didn't find an appreciable difference between this and, say, the `spark341db` versions of the classes.

`GpuDeltaCatalog.scala` is different for Databricks 14.3, I understand.

Reply:
(I ask because of [this comment](https://github.com/NVIDIA/spark-rapids/pull/12048/files/5b3aaf2ce913de2b4d94bc7fe6f4d7fc2779b106#r1935248976):
> This file is the same in every version of Databricks except for Databricks 14.3

Reply:
Line 50 is different. There may be a much smarter way to do this